exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/InternalTraceExporter.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2023 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.trace;
+
+import static com.google.api.client.util.Preconditions.checkNotNull;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcTransportChannel;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.trace.v2.TraceServiceClient;
+import com.google.cloud.trace.v2.TraceServiceSettings;
+import com.google.cloud.trace.v2.stub.TraceServiceStub;
+import com.google.common.collect.ImmutableMap;
+import com.google.devtools.cloudtrace.v2.AttributeValue;
+import com.google.devtools.cloudtrace.v2.ProjectName;
+import com.google.devtools.cloudtrace.v2.Span;
+import io.grpc.ManagedChannelBuilder;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class encapsulates internal implementation details for exporting spans for applications
+ * running on Google cloud environment. This class should not be exposed to the end user for direct
+ * usage.
+ */
+class InternalTraceExporter implements SpanExporter {
+
+  private final CloudTraceClient cloudTraceClient;
+  private final ProjectName projectName;
+  private final String projectId;
+  private final TraceTranslator translator;
+
+  private static final Map<String, String> HEADERS =
+      Map.of("User-Agent", "opentelemetry-operations-java/" + TraceVersions.EXPORTER_VERSION);
+  private static final HeaderProvider HEADER_PROVIDER = () -> HEADERS;
+
+  private static InternalTraceExporter createWithClient(
+      String projectId,
+      CloudTraceClient cloudTraceClient,
+      ImmutableMap<String, String> attributeMappings,
+      Map<String, AttributeValue> fixedAttributes) {
+    return new InternalTraceExporter(
+        projectId, cloudTraceClient, attributeMappings, fixedAttributes);
+  }
+
+  static SpanExporter createWithConfiguration(TraceConfiguration configuration) throws IOException {
```

Comment:
Consider wrapping the IOException at this layer and removing it from the signature. Since our external API no longer has an IOException, this just adds noise to our implementation.

Reply:
So this exception may be thrown from two possible places, handling it at the `InternalTraceExporter` level makes the implementation a little ugly since both need to be handled. Also, `InternalTraceExporter` is the class that actually has the implementation so handling it there would introduce more noise to our implementation detail. 

I think it makes sense to keep it in the function signature here since it has a package private access and this exception is handled in `TraceExporter` for both cases.